### PR TITLE
Initial commit of build container.

### PR DESCRIPTION
**Pull request by davidewatson**

### CODE REVIEW

File: container/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#ENV GOLANG_SHA256 b58bf5cede40b21812dfa031258db18fc39746cc0972bc26dae0393acc377aaf
+ENV GOLANG_SHA256 47fda42e46b4c3ec93fa5d4d4cc6a748aa3f9411a2a2b7e08e3a6d80d753ec8b
+
+ENV GLIDE_VERSION v0.12.3
+ENV GLIDE_URL https://github.com/Masterminds/glide/releases/download/$GLIDE_VERSION/glide-$GLIDE_VERSION-linux-amd64.tar.gz
+ENV GLIDE_SHA256 0e2be5e863464610ebc420443ccfab15cdfdf1c4ab63b5eb25d1216900a75109
+
+ENV GOSU_VERSION 1.9
+ENV GOSU_URL https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-amd64
+ENV GOSU_PATH /usr/local/bin/gosu
```
Comment (davidewatson) [inline container/Dockerfile]: Perhaps I should rename this GOSU_BINARY

File: container/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM buildpack-deps:jessie-scm
+
+#ENV GOLANG_VERSION 1.6.4
```
Comment (coffeepac) [inline container/Dockerfile]: please delete these commented out vars

**Comment by davidewatson** [comment]
I will rebase this off of the official image.

The reason was that we require cmake and their Dockerfile file does ```	&& rm -rf /var/lib/apt/lists/*``` before we get a chance to install cmake. However, I agree with you that basing this off of https://hub.docker.com/_/golang/ might simplify things. Fixing up apt to install cmake is not a big deal. And at the very least we would not have to update our Dockerfile with new sha256 hashes for golang each time we move to a new version.

File: container/Dockerfile (hunk 2 above)
Comment (davidewatson) [inline container/Dockerfile]: Will do.

**Comment by davidewatson** [comment]
Docker image rebased of golang:1.7.4

**Pull request merged**